### PR TITLE
Fix credentials cache issue

### PR DESCRIPTION
**Pull request by GeertvanHorrik**
This issue makes it impossible to provide different credentials for different feeds on the same subdomain, e.g.:

- https://extensions.wildgums.com/sub-url/nuget/index.json
- https://extensions.wildgums.com/another-sub-url/nuget/index.json

### CODE REVIEW

**Comment by GeertvanHorrik** [comment]
Please don't merge via GH, we will merge manually via tools since this hotfix needs to be merged to both develop and master.

This fix could lead to required re-authentication on existing software that uses this library.

File: src/Orc.NuGetExplorer.Tests/Services/CredentialsKeyHelperFacts.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    public class DummyCredentialsProvider : ICredentialProvider
+    {
+        public string Id => "dummy";
+
+        public Task<CredentialResponse> GetAsync(Uri uri, IWebProxy proxy, CredentialRequestType type, string message, bool isRetry, bool nonInteractive, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
```
Comment (Averus-a) [inline src/Orc.NuGetExplorer.Tests/Services/CredentialsKeyHelperFacts.cs:34]: Not implemented?

Comment (Averus-a) [inline src/Orc.NuGetExplorer.Tests/Services/CredentialsKeyHelperFacts.cs:34]: Not Implemented?

**Review by Averus-a — APPROVED** [review]

File: src/Orc.NuGetExplorer.Tests/Services/CredentialsKeyHelperFacts.cs (hunk 1 above)
Comment (GeertvanHorrik) [reply src/Orc.NuGetExplorer.Tests/Services/CredentialsKeyHelperFacts.cs:34]: It's a dummy implementation for unit testing the id.

**Pull request closed without merging**